### PR TITLE
[HelpDot] Add light mode

### PR DESCRIPTION
**Pull request by rushatgabhane**
### Fixed Issues

$ https://github.com/Expensify/App/issues/37911
PROPOSAL: N.A.

### Tests

1. Start helpsite  - `cd docs && npm run createDocsRoutes && bundle exec jekyll serve --livereload --safe`
2. Go to your computer's settings > appearance > Enable light mode.
3. Verify that the helpsite is in light mode.
4. Verify that the logo is black and green in color.
6. Navigate to an article and verify that text is following product color.
7. Go to search and verify that colors look good.

#### Dark mode
1. Enable dark mode from your computer's settings.
2. Verify that the helpsite is in dark mode.
3. Verify that logo is white and green.
4. Verify that production helpsite colors match for articles, search etc.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Light mode</summary>
### Home
image

### Footer
image

### Search
image

### Search result
image

### Hubs
image

### Sections
image

### Article 

image

### Platform selector

image

image

### Links

image

</details>

<details>
<summary>Dark mode</summary>
image

image

image

</details>

### CODE REVIEW

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:23]: I think rather than making these colors dependent on light theme, I would just make them the default. Then we only toggle into dark mode colors via the dark mode media query. But no need to put this one in a media query because we at least need a default at all times.

**Comment by rushatgabhane** [comment]
@Pujan92 could you please approve to assign internal. Thanks 🙇

**Comment by rushatgabhane** [comment]
@shawnborton I've attached screenshot for light and dark mode. Please let me know if it looks good 🙇

File: docs/assets/images/expensify-help.svg
Comment (rushatgabhane) [inline docs/assets/images/expensify-help.svg:1]:
```css
Added a media query here.

<style type="text/css">
    .st0{
         fill:#002e22
    }
    @media(prefers-color-scheme: dark){
        .st0{
            fill:#fffaf0
        }
    }
    .st1{
        fill:#03d47c
    }
</style>
```

**Comment by Pujan92** [comment]
> @Pujan92 could you please approve to assign internal. Thanks 🙇

Ok.

**Review by Pujan92 — APPROVED** [review]

**Review by Pujan92 — APPROVED** [review]

**Comment by tylerkaraszewski** [comment]
Tests are currently failing.

**Comment by rushatgabhane** [comment]
@tylerkaraszewski tests are expected to fail because the workflow doesn't have access to my fork

**Comment by tylerkaraszewski** [comment]
I'm curious how it's supposed to pass before we merge then.

**Comment by rushatgabhane** [comment]
Unfortunately `Deploy ExpensifyHelp` cannot pass on forks. So we have to test this locally by running the command in test steps

Or.. you could create a branch in App repo from this PR and see if tests pass?

**Comment by tylerkaraszewski** [comment]
Ok, I will do the review and test locally shortly.

**Comment by tylerkaraszewski** [comment]
CSS looks good but the logo is the wrong color for me.
Screenshot 2024-03-22 at 15 38 19

**Comment by rushatgabhane** [comment]
@tylerkaraszewski i verified on safari, chrome and incognito. Any ideas on how we can reproduce this?
The default colors for svg are in light mode - [related code here](https://github.com/Expensify/App/pull/38107#discussion_r1534686053)

Maybe run the command again? `cd docs && npm run createDocsRoutes && bundle exec jekyll serve --livereload --safe`

**Comment by rushatgabhane** [comment]
Another option is to hack our way to make `Deploy ExpenisfyHelp` work.
1. Because you have the permissions, you should create a new branch from `rushatgabhane:light-mode`
2. Create a PR to main
3. The workfow will work and deploy a preview of website

**Comment by tylerkaraszewski** [comment]
Ah, this was my fault. I had `main` checked out, ran the command, then realized I was on the wrong branch, and checked out `light-mode` but didn't run `npm run createDocsRoutes` again. It's working now. Screenshot 2024-03-22 at 18 16 10

**Comment by tylerkaraszewski** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by tylerkaraszewski — DISMISSED** [review]

**Comment by tylerkaraszewski** [comment]
I think this is good to go. Shall we wait for @shawnborton's approval?

**Comment by rushatgabhane** [comment]
Yes we should wait for @shawnborton's approval

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:18]: This value looks incorrect. I think it should be `rgba(235, 230, 223, 0.72)`

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:3]: Button text should be `FCFBF9`

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:12]: This should be `0164BF`

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:13]: Should be `0676DE`

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:24]: Should be `E7ECE9`

File: docs/_sass/_colors.scss
Comment (shawnborton) [inline docs/_sass/_colors.scss:34]: Should be `8DC8FF`

File: docs/_sass/_main.scss
Comment (shawnborton) [inline docs/_sass/_main.scss:136]: Weird that we aren't using a normal font size here. I know this isn't part of your PR but maybe we can follow up with some adjustments.

**Comment by shawnborton** [comment]
Is there a way to make a test build for easier testing?

And how are you making the logo change across light and dark modes?

File: docs/assets/images/expensify-help.svg
Comment (shawnborton) [reply docs/assets/images/expensify-help.svg:1]: Ah, nice. Didn't know you can do that! 

Can we change the fill color for the wordmark in dark mode to be `E7ECE9` please? Thanks!

File: docs/_sass/_main.scss
Comment (rushatgabhane) [reply docs/_sass/_main.scss:136]: yeah sure, created an issue - https://github.com/Expensify/App/issues/38986

**Comment by rushatgabhane** [comment]
@shawnborton i deployed my branch here - https://light-mode--fastidious-unicorn-246321.netlify.app/

You can use it to preview the website.

**Review by shawnborton — DISMISSED** [review]

**Comment by shawnborton** [comment]
Looks great to me!

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by rushatgabhane** [comment]
expected. the helpdot workflow doesn't work on forks

**Pull request merged**
